src/internal/components/dropdown-status/index.tsx

```diff
@@ -64,23 +72,35 @@ export const useDropdownStatus: UseDropdownStatus = ({
   isNoMatch,
   noMatch,
   onRecoveryClick,
+  recoveryProps,
 }) => {
   const linkRef = useRef<LinkProps.Ref | null>(null);
   const focusRecoveryLink = () => linkRef.current?.focus();
   const previousStatusType = usePrevious(statusType);
-
   const statusResult: DropdownStatusResult = { isSticky: true, content: null, focusRecoveryLink };
 
   if (statusType === 'loading') {
     statusResult.content = <InternalStatusIndicator type={'loading'}>{loadingText}</InternalStatusIndicator>;
   } else if (statusType === 'error') {
     statusResult.content = (
-      <>
+      <span
+        ref={recoveryProps ? recoveryProps.ref : null}
+        onBlur={
+          recoveryProps?.onBlur &&
```

Comment:
This condition is not needed, since `fireCancelableEvent` can also take `undefined` as a handler (and has this condition internally)